### PR TITLE
60 object blocking path

### LINKED ISSUE #60: Add ultrasonic code for object blocking path
Add code to the sensor robot so it can stop when an object is blocking the path.

### PR DESCRIPTION
**Pull request by nBukaty**
Added in code to detect objects on the path and prompt our robot to stop.  This also uses the smoothing library I created and tested earlier.

I used the ultrasonic sensor because we have tested it but this could easliy be replaced by one or more IR sensors.

resolves #60

### CODE REVIEW

File: arduino/sensors/linefollower/SensorArduino.ino [hunk 1]
```diff
@@ -60,20 +72,26 @@ void loop() {
     prevState = IDLE_STATE;
     break;
   case READ_LINE:
-    if(mySensorBar.getDensity() < 8 && mySensorBar.getDensity() > 0) {
-      nextState = GO_FORWARD;
-      if(mySensorBar.getPosition() < -50) {
-        nextState = GO_LEFT;
+    if(ultraSonic() > MAX_OBSTACLE_DISTANCE) {
```
Comment (nBukaty) [inline arduino/sensors/linefollower/SensorArduino.ino]: The outermost if else is the only real change here.

File: arduino/sensors/linefollower/SensorArduino.ino [hunk 2]
```diff
@@ -60,20 +72,26 @@ void loop() {
...
+        nextState = findLine();
+      }
+      else {
+        nextState = IDLE_STATE;
       }
-    }
-    else if(mySensorBar.getDensity() == 0) {
-      nextState = findLine();
     }
     else {
```
Comment (nBukaty) [inline arduino/sensors/linefollower/SensorArduino.ino]: If no object run as normal else stop motors.

File: arduino/sensors/linefollower/SensorArduino.ino [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #include <Wire.h>
 #include "sensorbar.h"
+#include "smoothing.h" //not sure how we're including with the make file
```
Comment (Amertz08) [inline arduino/sensors/linefollower/SensorArduino.ino]: I think we have to use the `-libraries` flag. I'll figure it out later.

Comment (Amertz08) [reply arduino/sensors/linefollower/SensorArduino.ino]: Where do you get "sensorbar.h"? is that a 3rd party install?

Comment (nBukaty) [reply arduino/sensors/linefollower/SensorArduino.ino]: Ya it's the library for the sensor bar. I'll see if I can add it in.

Comment (Amertz08) [reply arduino/sensors/linefollower/SensorArduino.ino]: Can it be installed via the Arduino library manager? I just might need to install them.

Comment (nBukaty) [reply arduino/sensors/linefollower/SensorArduino.ino]: I don't know I think I just downloaded it and placed it in the libraries folder.
https://github.com/sparkfun/SparkFun_Line_Follower_Array_Arduino_Library

**Comment by Amertz08** [comment]
@cduddy is this approved?

**Comment by Amertz08** [comment]
@cduddy

**Pull request merged**